src/Data/Enum.purs

```diff
@@ -37,10 +37,12 @@ derive newtype instance ordCardinality :: Ord (Cardinality a)
 -- | Type class for enumerations.
 -- |
 -- | Laws:
--- | - `succ a > pred a`
--- | - `pred a < succ a`
--- | - `pred >=> succ >=> pred = pred`
--- | - `succ >=> pred >=> succ = succ`
+-- | - Successor: `maybe true (a < _) (succ a)`
+-- | - Predecessor: `maybe true (_ < a) (pred a)`
+-- | - Succ retracts pred: `pred a >>= succ >>= pred = pred a`
+-- | - Pred retracts succ: `succ a >>= pred >>= succ = succ a`
+-- | - Non-skipping succ: `b <= a || maybe false (_ <= b) (succ a)`
```

Comment:
`forAllA` is a bit strange to me because the name doesn't really match up with what it does. For me, it obscures rather than illuminates.

Just an observation: your `given` is similar to `implies` from the `HeytingAlgebra Boolean` instance (which is in Prelude), so I might use that instead.

If `maybe` is awkward, try imagining it with the definition inlined:

```purescript
maybe x f y
-- becomes
case y of
  Just z -> f z
  Nothing -> x
```

I think it's probably best that the checks in quickcheck-laws match as closely as is reasonable with the actual laws in the documentation, though; in this case I think I would stick with the way they're written.